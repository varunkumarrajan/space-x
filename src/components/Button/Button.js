// React
import React from 'react';
// Prop types
import PropTypes from 'prop-types'; // ES6
// Styled component
import styled from 'styled-components';

// Styled button wrapper
const ButtonWrapper = styled.button`
  background-color: ${(props) => (props.active ? '#5aa55a' : '#afe3af')};
  display: inline-block;
  margin: 8px 18px;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  border: 1px solid transparent;
  padding: 4px 20px;
  font-size: 14px;
  line-height: 1.3;
  border-radius: 5px;
  width: auto;
  &:hover {
    background-color: #5aa55a;
  }
  &:focus {
    outline: 0;
  }
`;

/**
 * @description html button with extended style with styled component
 * @param {object} { button, handleButtonClick }
 */
const Button = ({ button, handleButtonClick }) => (
  <ButtonWrapper
    data-testid="buttonId"
    active={
      button.defaultValue
        ? button.defaultValue.split(',').includes(button.key)
        : false
    }
    onClick={() => handleButtonClick(button.key, button.type)}
  >
    {button.value}
  </ButtonWrapper>
);

// Default export
export default Button;

// Default props
Button.defaultProps = {
  button: {
    defaultValue: '',
    key: '',
    value: '',
    type: '',
  },
  handleButtonClick: () => {},
};

// Prop types
Button.propTypes = {
  button: PropTypes.shape({
    defaultValue: PropTypes.string,
    key: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }),
  handleButtonClick: PropTypes.func.isRequired,
};
