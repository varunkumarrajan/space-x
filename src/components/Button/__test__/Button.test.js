// React
import React from 'react';
// Testing library
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Button from './../Button';

// Mock data
const buttonMock = {
  defaultValue: 'test',
  key: 'test',
  value: 'test',
  type: 'test',
}
const buttonMockFunc = () => {};

afterEach(cleanup);

describe('Button', () => {
  it('Should Render Button With Props', () => {
    const { getByTestId } = render(<Button button={buttonMock} handleButtonClick={buttonMockFunc} />);
    const buttonId = getByTestId('buttonId');
    const element = screen.getByText(/test/i);
    fireEvent.click(buttonId, buttonMockFunc);
    expect(element).toBeInTheDocument();
  });

  it('Button Snapshot', () => {
    const button = renderer.create(<Button  button={buttonMock} handleButtonClick={buttonMockFunc} />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
