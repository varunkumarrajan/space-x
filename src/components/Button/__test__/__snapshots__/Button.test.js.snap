// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button Snapshot 1`] = `
<button
  className="sc-bdfBwQ gXAVEs"
  data-testid="buttonId"
  onClick={[Function]}
>
  test
</button>
`;
